### ISSUE TITLE
do you use gradient_accumulation_steps when training?

### ISSUE THREAD

**Issue by lijiaoyang** [TIGER-AI-Lab/VLM2Vec#50]
Dear author, I see your paper said: When using GradCache, we set a sub-batch size of 4 to enable full model tuning, with the **total batch size** accumulated to 1,024. your batchsize is 256, which means: 1024 = 256 * 4
According to [the code](https://github.com/TIGER-AI-Lab/VLM2Vec/blob/main/src/trainer.py#L262) , GradCache sub-batch is not used when computing **total_train_batch_size**. 
May I ask why this is? Do you use the common parameter **gradient_accumulation_steps** in your training step?

**Comment by XMHZZ2018** [comment]
Hi @lijiaoyang ,
Thanks for your interest in our work!
The global batch size is determined by ```--per_device_train_batch_size``` multiplied by the number of GPUs. So if there are 8 gpus and total batch size of 1024, you can use ```--per_device_train_batch_size``` 128.
GradCache sub-batch is used to avoid OOM and maximize the gpu utility, and it's not related to total batch size.
We did not use ```gradient_accumulation_steps``` since it does not help increase the number of random negatives in contrastive loss. We have provided the training parameter setup here: https://github.com/TIGER-AI-Lab/VLM2Vec/blob/main/examples/qwen2vl/run_train.sh

Please let me know if this answer helps address your question. Thanks!
